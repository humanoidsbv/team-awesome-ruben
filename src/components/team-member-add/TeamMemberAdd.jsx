import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';

import styles from './TeamMemberAdd.module.css';
import UserProfileImage from '../../../static/images/picture-tnt.jpg';

const TeamMemberAdd = ({ addFormData, handleFormVisible }) => {
  const [bio, setBio] = useState('');
  const [emailAdress, setEmailAdress] = useState('');
  const [facebook, setFacebook] = useState('');
  const [firstName, setFirstName] = useState('');
  const [homeAdress, setHomeAdress] = useState('');
  const [lastName, setLastName] = useState('');
  const [linkedIn, setLinkedIn] = useState('');
  const [locality, setLocality] = useState('');
  const [postalCode, setPostalCode] = useState('');

  const [validity, setValidity] = useState({});

  const formRef = useRef(null);

  const handleBlur = event => {
    setValidity({
      ...validity,
      form: formRef.current.checkValidity(),
      [event.target.name]: event.target.checkValidity()
    });
  };

  const handleCloseForm = () => {
    handleFormVisible();
    setBio('');
    setEmailAdress('');
    setFacebook('');
    setFirstName('');
    setHomeAdress('');
    setLastName('');
    setLinkedIn('');
    setLocality('');
    setPostalCode('');
  };

  const handleSubmit = event => {
    event.preventDefault();

    addFormData({
      bio,
      currentClient: '-',
      employeeNumber: '-',
      firstName,
      homeAdress,
      id: Math.random(),
      lastName,
      locality,
      role: '-',
      startingDate: '-',
      postalCode
    });

    handleCloseForm();
  };

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <span className={styles.heading}>Add new team member</span>
        <button
          className={styles.cancelButton}
          onClick={handleCloseForm}
          type="button"
        >
          Cancel
        </button>
        <button
          className={styles.addMemberButton}
          disabled={validity.form !== true}
          onClick={handleSubmit}
          type="button"
        >
          Save
        </button>
      </div>
      <div className={styles.formTab}>
        <span>Personal Details</span>
      </div>
      <form className={styles.form} onSubmit={handleSubmit} ref={formRef}>
        <div className={styles.profileImageContainer}>
          <img
            alt="Profile avatar"
            className={styles.profileImage}
            src={UserProfileImage}
          />
          <span>Edit Avatar</span>
        </div>
        <div className={styles.inputContainer}>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="firstName"
          >
            <span className={styles.labelText}>First name</span>
            <input
              className={`${styles.input} ${
                validity.firstName === false ? styles.invalidInput : ''
              }`}
              maxLength="30"
              minLength="2"
              name="firstName"
              onBlur={handleBlur}
              onChange={({ target }) => setFirstName(target.value)}
              required
              value={firstName}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="lastName"
          >
            <span className={styles.labelText}>Last name</span>
            <input
              className={`${styles.input} ${
                validity.lastName === false ? styles.invalidInput : ''
              }`}
              maxLength="30"
              minLength="2"
              name="lastName"
              onBlur={handleBlur}
              onChange={({ target }) => setLastName(target.value)}
              required
              value={lastName}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="emailAdress">
            <span className={styles.labelText}>Email adress</span>
            <input
              className={`${styles.input} 
              ${validity.emailAdress === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="emailAdress"
              onBlur={handleBlur}
              onChange={({ target }) => setEmailAdress(target.value)}
              required
              value={emailAdress}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="bio">
            <span className={styles.labelText}>Bio</span>
            <textarea
              className={`${styles.input} ${styles.textArea}`}
              name="bio"
              onBlur={handleBlur}
              onChange={({ target }) => setBio(target.value)}
              value={bio}
            />
          </label>
        </div>
        <div className={styles.inputContainer}>
          <label className={`${styles.label}`} htmlFor="homeAdress">
            <span className={styles.labelText}>Adress</span>
            <input
              className={`${styles.input} 
              ${validity.homeAdress === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="homeAdress"
              onBlur={handleBlur}
              onChange={({ target }) => setHomeAdress(target.value)}
              required
              value={homeAdress}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="postalCode"
          >
            <span className={styles.labelText}>Postal code</span>
            <input
              className={`${styles.input} 
              ${validity.postalCode === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="postalCode"
              onBlur={handleBlur}
              onChange={({ target }) => setPostalCode(target.value)}
              required
              value={postalCode}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="locality"
          >
            <span className={styles.labelText}>Place of birth</span>
            <input
              className={`${styles.input} 
              ${validity.locality === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="locality"
              onBlur={handleBlur}
              onChange={({ target }) => setLocality(target.value)}
              required
              value={locality}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="socialProfiles">
            <span className={styles.labelText}>Social profiles</span>
            <input
              className={`${styles.input} ${styles.facebook}
              ${validity.facebook === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="faceBook"
              onBlur={handleBlur}
              onChange={({ target }) => setFacebook(target.value)}
              value={facebook}
            />
            <input
              className={`${styles.input}
              ${validity.linkedIn === false ? styles.invalidInput : ''}`}
              maxLength="30"
              minLength="2"
              name="linkedIn"
              onBlur={handleBlur}
              onChange={({ target }) => setLinkedIn(target.value)}
              required
              value={linkedIn}
            />
          </label>
        </div>
      </form>
    </div>
  );
};

TeamMemberAdd.propTypes = {
  addFormData: PropTypes.func.isRequired,
  handleFormVisible: PropTypes.func.isRequired
};

export default TeamMemberAdd;
