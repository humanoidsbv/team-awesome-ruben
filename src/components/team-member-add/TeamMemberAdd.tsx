import React, { useState, useRef } from 'react';

import styles from './TeamMemberAdd.module.css';
import UserProfileImage from '../../../static/images/picture-tnt.jpg';

import { ValidityState } from '../../../shared/types';

interface TeamMemberProps {
  //* TODO: Add data to addFormData and rename for team members
  addFormData: ({}) => void;
  handleFormVisible: () => void;
}

const TeamMemberAdd = ({
  addFormData,
  handleFormVisible
}: TeamMemberProps): React.ReactElement => {
  const [bio, setBio] = useState('');
  const [emailaddress, setEmailaddress] = useState('');
  const [facebook, setFacebook] = useState('');
  const [firstName, setFirstName] = useState('');
  const [homeaddress, setHomeaddress] = useState('');
  const [lastName, setLastName] = useState('');
  const [linkedIn, setLinkedIn] = useState('');
  const [locality, setLocality] = useState('');
  const [postalCode, setPostalCode] = useState('');

  const [validity, setValidity] = useState<ValidityState>({});

  const formRef = useRef(null);

  const handleBlur = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void => {
    setValidity({
      ...validity,
      form: formRef.current.checkValidity(),
      [event.target.name]: event.target.checkValidity()
    });
  };

  const handleCloseForm = (): void => {
    handleFormVisible();
    setBio('');
    setEmailaddress('');
    setFacebook('');
    setFirstName('');
    setHomeaddress('');
    setLastName('');
    setLinkedIn('');
    setLocality('');
    setPostalCode('');
  };

  const handleSubmit = (event: React.ChangeEvent<HTMLInputElement>): void => {
    event.preventDefault();

    addFormData({
      bio,
      currentClient: '-',
      employeeNumber: '-',
      firstName,
      homeaddress,
      id: Math.random(),
      lastName,
      locality,
      role: '-',
      startingDate: '-',
      postalCode
    });

    handleCloseForm();
  };

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <span className={styles.heading}>Add new team member</span>
        <button
          className={styles.cancelButton}
          onClick={handleCloseForm}
          type="button"
        >
          Cancel
        </button>
        <button
          className={styles.addMemberButton}
          disabled={validity.form !== true}
          onClick={() => handleSubmit}
          type="button"
        >
          Save
        </button>
      </div>
      <div className={styles.formTab}>
        <span>Personal Details</span>
      </div>
      <form className={styles.form} onSubmit={() => handleSubmit} ref={formRef}>
        <div className={styles.profileImageContainer}>
          <img
            alt="Profile avatar"
            className={styles.profileImage}
            src={UserProfileImage}
          />
          <span>Edit Avatar</span>
        </div>
        <div className={styles.inputContainer}>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="firstName"
          >
            <span className={styles.labelText}>First name</span>
            <input
              className={`${styles.input} ${validity.firstName === false &&
                styles.invalidInput}`}
              maxLength={30}
              minLength={2}
              name="firstName"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setFirstName(target.value)
              }
              required
              value={firstName}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="lastName"
          >
            <span className={styles.labelText}>Last name</span>
            <input
              className={`${styles.input} ${validity.lastName === false &&
                styles.invalidInput}`}
              maxLength={30}
              minLength={2}
              name="lastName"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setLastName(target.value)
              }
              required
              value={lastName}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="emailaddress">
            <span className={styles.labelText}>Email address</span>
            <input
              className={`${styles.input} 
              ${validity.emailaddress === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="emailaddress"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setEmailaddress(target.value)
              }
              required
              value={emailaddress}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="bio">
            <span className={styles.labelText}>Bio</span>
            <textarea
              className={`${styles.input} ${styles.textArea}`}
              name="bio"
              onBlur={handleBlur}
              onChange={({ target }): void | string => setBio(target.value)}
              value={bio}
            />
          </label>
        </div>
        <div className={styles.inputContainer}>
          <label className={`${styles.label}`} htmlFor="homeaddress">
            <span className={styles.labelText}>address</span>
            <input
              className={`${styles.input} 
              ${validity.homeaddress === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="homeaddress"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setHomeaddress(target.value)
              }
              required
              value={homeaddress}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="postalCode"
          >
            <span className={styles.labelText}>Postal code</span>
            <input
              className={`${styles.input} 
              ${validity.postalCode === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="postalCode"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setPostalCode(target.value)
              }
              required
              value={postalCode}
            />
          </label>
          <label
            className={`${styles.label} ${styles.labelHalfWidth}`}
            htmlFor="locality"
          >
            <span className={styles.labelText}>Place of birth</span>
            <input
              className={`${styles.input} 
              ${validity.locality === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="locality"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setLocality(target.value)
              }
              required
              value={locality}
            />
          </label>
          <label className={`${styles.label}`} htmlFor="socialProfiles">
            <span className={styles.labelText}>Social profiles</span>
            <input
              className={`${styles.input} ${styles.facebook}
              ${validity.facebook === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="faceBook"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setFacebook(target.value)
              }
              value={facebook}
            />
            <input
              className={`${styles.input}
              ${validity.linkedIn === false ? styles.invalidInput : ''}`}
              maxLength={30}
              minLength={2}
              name="linkedIn"
              onBlur={handleBlur}
              onChange={({
                target
              }: React.ChangeEvent<HTMLInputElement>): void | string =>
                setLinkedIn(target.value)
              }
              required
              value={linkedIn}
            />
          </label>
        </div>
      </form>
    </div>
  );
};

export default TeamMemberAdd;
