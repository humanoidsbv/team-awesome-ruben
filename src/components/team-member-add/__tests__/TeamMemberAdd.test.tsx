import * as React from 'react';
import { shallow } from 'enzyme';

import TeamMemberAdd from '../TeamMemberAdd';

it('matches component output with expected output', () => {
  const teamMemberAdd = shallow(
    <TeamMemberAdd addFormData={() => {}} handleFormVisible={() => {}} />
  );
  expect(teamMemberAdd).toMatchSnapshot();
});
