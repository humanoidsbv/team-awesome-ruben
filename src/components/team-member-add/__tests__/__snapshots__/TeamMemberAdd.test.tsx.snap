// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <span
      className="heading"
    >
      Add new team member
    </span>
    <button
      className="cancelButton"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
    <button
      className="addMemberButton"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
  <div
    className="formTab"
  >
    <span>
      Personal Details
    </span>
  </div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="profileImageContainer"
    >
      <img
        alt="Profile avatar"
        className="profileImage"
        src="test-file-stub"
      />
      <span>
        Edit Avatar
      </span>
    </div>
    <div
      className="inputContainer"
    >
      <label
        className="label labelHalfWidth"
        htmlFor="firstName"
      >
        <span
          className="labelText"
        >
          First name
        </span>
        <input
          className="input false"
          maxLength={30}
          minLength={2}
          name="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label labelHalfWidth"
        htmlFor="lastName"
      >
        <span
          className="labelText"
        >
          Last name
        </span>
        <input
          className="input false"
          maxLength={30}
          minLength={2}
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="emailaddress"
      >
        <span
          className="labelText"
        >
          Email address
        </span>
        <input
          className="input 
              "
          maxLength={30}
          minLength={2}
          name="emailaddress"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="bio"
      >
        <span
          className="labelText"
        >
          Bio
        </span>
        <textarea
          className="input textArea"
          name="bio"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
      </label>
    </div>
    <div
      className="inputContainer"
    >
      <label
        className="label"
        htmlFor="homeaddress"
      >
        <span
          className="labelText"
        >
          address
        </span>
        <input
          className="input 
              "
          maxLength={30}
          minLength={2}
          name="homeaddress"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label labelHalfWidth"
        htmlFor="postalCode"
      >
        <span
          className="labelText"
        >
          Postal code
        </span>
        <input
          className="input 
              "
          maxLength={30}
          minLength={2}
          name="postalCode"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label labelHalfWidth"
        htmlFor="locality"
      >
        <span
          className="labelText"
        >
          Place of birth
        </span>
        <input
          className="input 
              "
          maxLength={30}
          minLength={2}
          name="locality"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="socialProfiles"
      >
        <span
          className="labelText"
        >
          Social profiles
        </span>
        <input
          className="input facebook
              "
          maxLength={30}
          minLength={2}
          name="faceBook"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <input
          className="input
              "
          maxLength={30}
          minLength={2}
          name="linkedIn"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;
